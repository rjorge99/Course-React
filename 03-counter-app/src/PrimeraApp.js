import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {
    return (
        <>
            {saludo}
            <p>{subtitulo}</p>
        </>
    );
};

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
};

PrimeraApp.defaultProps = {
    subtitulo: 'Informacion Default'
};

export default PrimeraApp;
