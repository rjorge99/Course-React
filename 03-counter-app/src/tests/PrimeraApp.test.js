import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';

describe('Pruebas en PrimeraApp', () => {
    // test('Debe de mostrar el mensaje Hola', () => {
    //     const saludo = 'Hola';
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // });

    test('debe mostrar PrimeraApp correctamente', () => {
        const saludo = 'Hola';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });
    test('debe mostrar el subtitulo', () => {
        const saludo = 'Hola';
        const subtitulo = 'soy subtitulo';
        const wrapper = shallow(
            <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
        );

        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);
    });
});
