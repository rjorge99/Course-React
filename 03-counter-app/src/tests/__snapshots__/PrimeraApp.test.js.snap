// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  Hola
  <p>
    Informacion Default
  </p>
</Fragment>
`;
